services/api/src/resources/environment/resolvers.ts

```diff
@@ -308,6 +309,71 @@ export const getEnvironmentByKubernetesNamespaceName: ResolverFn = async (
     ctx
   );
 
+export const getEnvironmentsByKubernetes: ResolverFn = async (
+  _,
+  { kubernetes, order, createdAfter, type },
+  { sqlClientPool, hasPermission, models, keycloakGrant }
+) => {
+  const openshift = await openshiftHelpers(
+    sqlClientPool
+  ).getOpenshiftByOpenshiftInput(kubernetes);
+
+  let userProjectIds: number[];
+  try {
+    await hasPermission('openshift', 'viewAll');
+  } catch (err) {
+    if (!keycloakGrant) {
+      logger.warn('No grant available for getEnvironmentsByKubernetes');
+      return [];
+    }
+
+    // Only return projects the user can view
+    userProjectIds = await models.UserModel.getAllProjectsIdsForUser({
```

Comment:
For non-admin users using this query, it fails because the way this works was changed in the keycloak optimisations PR.

You'll need to update this to use the following
```
const userProjectRoles = await models.UserModel.getAllProjectsIdsForUser({
  id: keycloakGrant.access_token.content.sub,
}, keycloakUsersGroups);
userProjectIds = getUserProjectIdsFromRoleProjectIds(userProjectRoles);
```

You'll also need to update the resolver function to include `keycloakUsersGroups` and also import 
```
import { getUserProjectIdsFromRoleProjectIds } from '../../util/auth';
```